python/cudf/cudf/core/column_accessor.py

```diff
@@ -207,11 +207,16 @@ def _from_columns_like_self(
 
     @property
     def level_names(self) -> tuple[abc.Hashable, ...]:
+        if self.is_cached("to_pandas_index"):
+            return self.to_pandas_index.names
```

Comment:
Is there any particular reason we want/need to use the cached `to_pandas_index` here?

Reply:
Yes, when updates to the pandas object created by `to_pandas_index` are performed, we need the updated `names`, that's what this piece of code does.